frontend/rolecall/src/app/api/user_api.service.ts

```diff
@@ -10,67 +10,51 @@ import { LoggingService } from '../services/logging.service';
 import { ResponseStatusHandlerService } from '../services/response-status-handler.service';
 
 export type User = {
-  uuid: APITypes.UserUUID,
-  has_permissions: APITypes.PermissionSet,
-  has_privilege_classes?: APITypes.PrivilegeClassUUID[],
-  knows_positions: APITypes.Position[],
-  first_name: string,
-  last_name: string,
-  date_joined: number,
-  role?: string,
+  uuid: APITypes.UserUUID;
+  has_roles: APITypes.RoleSet;
+  has_permissions: APITypes.PermissionSet;
+  knows_positions: APITypes.Position[];
+  first_name: string | undefined;
+  last_name: string | undefined;
+  date_joined: number | undefined;
   contact_info: {
-    phone_number: string,
-    email: string,
+    phone_number: string | undefined;
```

Comment:
The email is a required field. The phone number is not required now, but it could change going forward.
However, this data structure is now supporting user input, so it has to allow for undefined values. There is no field-is-defined check on the front end. That checking is done on the backend.